delta-lake/delta-spark341db/pom.xml

```diff
@@ -38,6 +38,10 @@
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.roaringbitmap</groupId>
+            <artifactId>RoaringBitmap</artifactId>
```

Comment:
This dependency is used in delta-lake/common and thus needs to be in every delta-lake shim pom, as we want that specific, shaded version when using Delta Lake.  There are other org.roaringbitmap uses (e,.g,.: RapidsPushBasedFetchHelper, RapidsShuffleBlockFetcher) but I think those can use the Spark version.  The Delta Lake usage require a minimum org.roaringbitmap version that may not be provided by a particular Spark platform.

Reply:
But according to the [PR](https://github.com/NVIDIA/spark-rapids/pull/10979) it only affects 341db and since delta-lake/common code doesn't have a POM and is only as source in respective delta-lake versions it will pick up the right dependency defined in 341db's POM.

Reply:
But this is very fragile now.  When someone adds code to Delta Lake that references this, it will _silently_ start using the Spark version.  Maybe that works, maybe it doesn't,.  Depends on what the underlying platform actually provides for the org.roaringbitmap version, but we need a specific version to work -- that's why it was originally placed where it was and shaded.  The Delta common code has a reference to this class, so it should be available in all Delta versions.

Reply:
Also note that delta24x/GpuLowShuffleMergeCommand is also using it, so it's not specific to the spark341db shim.  Let's put it in all the shims using the common code (all but the delta stub) or we're likely going to get surprised in a confusing way in the future.